### ISSUE TITLE
[FR] Saving Snippets in a Hidden Folder

### ISSUE THREAD

**Issue by zcysxy** [artisticat1/obsidian-latex-suite#67]
It would be great if the plugin can read a snippets file in a hidden directory like `.config/`

**Comment by windily-cloud** [comment]
I think it would be better to store snippets in the vault so I can edit the snippets at any time. but currently if you save the snippets as md files it can lead to confusing formatting.

I can add a feature to latex-suite that can open the snippets, I save it as a js file and it has syntax highlighting. Something like this.

<img width="865" alt="image" src="https://user-images.githubusercontent.com/30640638/196145366-01679d41-cb21-4741-b206-9ef546b9f65b.png">

**Comment by zcysxy** [comment]
Thanks for the detailed reply @windily-cloud! My `.config/` folder is actually in the vault, storing some config files for plugins like `.vimrc`. But yeah it's a good idea to store it in a non-hidden folder so you can edit it directly in Obsidian.

**Comment by artisticat1** [comment]
Obsidian does not appear to emit a `modify` event when a file in a hidden folder (such as `.config/`) is edited, so I'm not sure how I would watch for changes to the snippet file.

> I can add a feature to latex-suite that can open the snippets, I save it as a js file and it has syntax highlighting. Something like this. The code is not much, if @artisticat1 is willing to add it would be better.

That looks nice! You're welcome to submit a PR if you want. I'd be happy to accept it if the custom view only applies to the snippet file.

**Comment by zcysxy** [comment]
> Obsidian does not appear to emit a `modify` event when a file in a hidden folder (such as `.config/`) is edited, so I'm not sure how I would watch for changes to the snippet file.

You are right. Plugins like obsidian-vimrc that can read files in a hidden folder need reloading to take effect after editing.

**Comment by windily-cloud** [comment]
> That looks nice! You're welcome to submit a PR if you want. I'd be happy to accept it if the custom view only applies to the snippet file.

I submit a PR #69 .  Now I can edit and load snippets in real time in obsidian, code highlighting and vim mode are also supported.
